cli/JSXV4UPGRADE.md

```diff
@@ -0,0 +1,44 @@
+**JSX V4 upgrade path**
+
+1. No upgrading rescript-react && no changes in bsconfig.json
+
+   JSX V3 by default.
+
+2. No upgrading rescript-react && new configuration added in bsconfig.json
+
+   a. JSX V3
+
+   ```json
+   "react": {
+     "jsx": 3,
+     "runtime": "classic"
+   }
+   ```
+
+   The `react.runtime` affects nothing, no matter of `"classic"` or `"automatic"`.
+
+   b. JSX V4 with classic mode
+
+   ```json
+   "react": {
+     "jsx": 4,
+     "runtime": "classic"
+   }
+   ```
+
+   c. JSX V4 with new JSX mode (Experimental)
+
+   ```json
+   "react": {
+     "jsx": 4,
+     "runtime": "automatic"
+   }
+   ```
+
+   JSX V4 with `Js.React` which needs the peer-dependecy of React v17.\* or higher. It may break the project with dependencies which are using the explicit types of `rescript-react`.
```

Comment:
> `Js.React`

What's the idea of making it a submodule of `Js`? How is a user supposed to define this binding?
Also, do we again couple the JSX implementation to React? I thought we wanted to explore more liberate systems that will allow different targets.

Reply:
It is just a test try to add missing bindings of the new JSX mode which is wrongly bound. https://github.com/rescript-lang/syntax/pull/547#issuecomment-1166177098

Reply:
I guess what I rather wanted to see is a config model that is similar to [TS jsx option](https://www.typescriptlang.org/tsconfig#jsx)